driver/csiplugin/connectors/rest_v2.go

```diff
@@ -760,10 +848,82 @@ func (s *spectrumRestV2) GetFileSetNameFromId(filesystemName string, Id string)
 	return getFilesetResponse.Filesets[0].FilesetName, nil
 }
 
+func (s *spectrumRestV2) GetSnapshotCreateTimestamp(filesystemName string, filesetName string, snapName string) (string, error) {
+	glog.V(4).Infof("rest_v2 GetSnapshotCreateTimestamp. filesystem: %s, fileset: %s, snapshot: %s ", filesystemName, filesetName, snapName)
+
+	getSnapshotURL := utils.FormatURL(s.endpoint, fmt.Sprintf("scalemgmt/v2/filesystems/%s/filesets/%s/snapshots/%s", filesystemName, filesetName, snapName))
+	getSnapshotResponse := GetSnapshotResponse_v2{}
+
+	err := s.doHTTP(getSnapshotURL, "GET", &getSnapshotResponse, nil)
+	if err != nil {
+		return "", fmt.Errorf("unable to list snapshot %v", snapName)
+	}
+
+	if len(getSnapshotResponse.Snapshots) == 0 {
+		return "", fmt.Errorf("unable to list snapshot %v", snapName)
+	}
+
+	return fmt.Sprintf("%d", getSnapshotResponse.Snapshots[0].Created), nil
+}
+
+func (s *spectrumRestV2) GetSnapshotUid(filesystemName string, filesetName string, snapName string) (string, error) {
```

Comment:
Why not have a helper function that retrieves the Snapshot Response? Do we need four functions that perform the same REST query?

Reply:
Also do we need to make that many REST queries? Once we get the snapshot data can we assume we have the data and don't need to keep querying  the REST endpoint?

Reply:
All these functions are invoked from different execution paths and response received at one place cannot be reused at other. These are getting invoked from CreateSnapshot, DeleteSnapshot and DeleteVolume APIs.

rest_v2.go is a util class containing convenience functions, hence separate functions returning the specific value are desired instead of returning a common response and letting the caller parse it to fetch desired value. It doesnt meant same REST query gets invoked multiple times in the same execution.